homeassistant/components/zeroconf/__init__.py

```diff
@@ -37,11 +43,55 @@
 
     await zeroconf.register_service(info)
 
-    async def stop_zeroconf(event):
+    async def new_service(service_type, name):
+        """Signal new service discovered."""
+        service_info = await zeroconf.get_service_info(service_type, name)
+        info = info_from_service(service_info)
+        print(name, info)
+
+        for domain in manifest.SERVICE_TYPES[service_type]:
+            hass.async_create_task(
+                hass.config_entries.flow.async_init(
+                    domain, context={'source': DOMAIN}, data=info
+                )
+            )
+
+    def service_update(_, service_type, name, state_change):
+        """Service state changed."""
+        from aiozeroconf import ServiceStateChange
+
+        if state_change is ServiceStateChange.Added:
+            hass.async_create_task(new_service(service_type, name))
+
+    for service in manifest.SERVICE_TYPES:
+        ServiceBrowser(zeroconf, service, handlers=[service_update])
+
+    async def stop_zeroconf(_):
         """Stop Zeroconf."""
         await zeroconf.unregister_service(info)
         await zeroconf.close()
 
     hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, stop_zeroconf)
 
     return True
+
+
+def info_from_service(service):
+    """Return prepared info from mDNS entries."""
+    properties = {}
+
+    for key, value in service.properties.items():
+        if isinstance(value, bytes):
+            value = value.decode('utf-8')
```

Comment:
As mDNS uses UDP as a transport, this may fail if some bits get flipped. The question is: does it make sense to keep going (i.e. suppressing the error) or simply let this crash? I'd say it's better to avoid crash here, and let the integration developers to deal with potentially missing key, as this way the user gets at least feedback that device has been detected even when it doesn't work.

Reply:
This is the same code used in netdisco and there haven't been any real issues there. I'd say keep it as is

Reply:
I do like defensive programming and catching bad values, exceptions that are raised might kill the whole process. Better to be able to warn about the bad device.

Reply:
Better?